### DISCUSSION TITLE
Sysex not being received for a DX21 synth

### DISCUSSION THREAD

**Discussion by alexlipscomb** [djipco/webmidi#227] [Q&A]
Hey all,
I'm writing a patch storage app in Electron that receives and saves bulk data transfers from a DX21 synthesizer. I can't seem to receive any sysex through WebMidi.js when I transmit from the synth. I'm able to receive it, however, on applications such as Max/MSP and through the Python RTMidi library with no problems at all.

Here is the code that I'm using to listen for sysex:
```ts
import {WebMidi} from 'webmidi';

WebMidi.enable({sysex: true}).then(listenForSysex);

// ...

function listenForSysex(): void {
    WebMidi.getInputByName(selectedDeviceName).addOneTimeListener('sysex', (e: MessageEvent) => { // Never reaches here
        parseSysex(e);
    }
}

```

I'm also unable to receive this sysex when passing it through Max/MSP or my Python implementation first, even though it gets received and printed to the console in both situations.

Any help on this would be great, or pointers to find a workaround.

macOS Catalina 10.15.7, running on Electron 16.0.7 with Typescript.

**Comment by djipco** [comment]
There is a unit test to test reception of `sysex` messages. So, it is supposed to work. However, TypeScript support is new in version 3 and the problem might be related to that.

Would you be able to test the same code in vanilla JavaScript (no TypeScript) to see if the problem still is present?

**Reply by alexlipscomb** [reply]
Tested in vanilla JavaScript with the same code, still doesn't work. Curious about that since the sysex works perfectly fine with any other synth I own. Only the DX21 and WebMidi play together poorly, nothing else.

Edit: Note Messages and CC work perfectly. It's only 32 Bulk Voice Data and 1 Bulk Voice Data as far as I'm aware.

**Reply by djipco** [reply]
This is odd.

I guess the logical next step would be to identify whether the problem is at the WEBMIDI.js level or upstream at the Web MIDI API level.

Would you be able to test this bare Web MIDI API code (below)?

```js
navigator.requestMIDIAccess({sysex: true}).then(onMIDISuccess, onMIDIFailure);

function onMIDISuccess(midiAccess) {

    console.log("sysex enabled: ", midiAccess.sysexEnabled);

    for (var input of midiAccess.inputs.values())
        input.onmidimessage = message => console.log(message);
   }

}

function onMIDIFailure() {
    console.log('Could not access your MIDI devices.');
}
```
I would like to get at the bottom of this.

Cheers!
